tests/test_lexer.py

```diff
@@ -1,116 +1,118 @@
-from mplee_interpreter.lexer import Lexer
-from mplee_interpreter.definitions.values import Token
+from mplee_interpreter.definitions.types import (TT_EOF, TT_EQ, TT_GTE, TT_INT,
+                                                 TT_KEYWORD, TT_LTE, TT_MINUS,
+                                                 TT_NE, TT_STRING)
 from mplee_interpreter.definitions.utils import Position
-from mplee_interpreter.definitions.types import (
-    TT_INT,
-    TT_EOF,
-    TT_STRING,
-    TT_KEYWORD,
-    TT_MINUS,
-    TT_NE,
-    TT_EQ,
-    TT_LTE,
-    TT_GTE
-)
-import pytest
-
-@pytest.mark.parametrize("mplee", ["5"])
-def test_lexer_creates_tokens(lexer_make_tokens_return_tokens):
-    assert isinstance(lexer_make_tokens_return_tokens[0], Token)
-    assert isinstance(lexer_make_tokens_return_tokens[1], Token)
-    
-@pytest.mark.parametrize("mplee", ["5"])
-def test_is_first_token_value_5(lexer_make_tokens_return_tokens):
-    result = lexer_make_tokens_return_tokens
-    assert isinstance(result[0].value, int)
-
-@pytest.mark.parametrize("mplee", ["5"])
-def test_advance(lexer_object):
+from mplee_interpreter.definitions.values import Token
+from mplee_interpreter.lexer import Lexer
+
+
+def test_checks_if_make_tokens_method_return_tokens():
+    lexer = Lexer("example.mplee", "5")
+    tokens, _ = lexer.make_tokens()
+    assert isinstance(tokens[0], Token)
+    assert isinstance(tokens[1], Token)
+
+
+def test_if_value_of_first_token_is_5_for_input_5():
+    lexer = Lexer("example.mplee", "5")
+    tokens, _ = lexer.make_tokens()
+    assert isinstance(tokens[0].value, int)
+
+
+def test_if_id_line_and_column_will_change_their_value_correctly_in_Position_class():
+    lexer = Lexer("example.mplee", "5")
     pos = Position(1, 0, 1, "mplee", "5")
-    lexer_object_ref = lexer_object
+    lexer_object_ref = lexer
     lexer_object_ref.advance()
-    assert pos.idx == lexer_object_ref.pos.idx   
-    assert pos.ln == lexer_object_ref.pos.ln 
+    assert pos.idx == lexer_object_ref.pos.idx
+    assert pos.ln == lexer_object_ref.pos.ln
     assert pos.col == lexer_object_ref.pos.col
 
-@pytest.mark.parametrize("mplee", ["5"])
-def test_make_tokens(lexer_make_tokens_return_tokens):
-    assert isinstance(lexer_make_tokens_return_tokens[0], Token)
-    assert isinstance(lexer_make_tokens_return_tokens[1], Token)
-    
+
+def test_if_make_tokens_method_return_EOF_token_for_input_5():

```

Comment:
```suggestion
def test_make_tokens_returns_EOF_token_if_....():

```

Ja bym nazywał testy na zasadzie test_<nazwa metody/klasy>_<co_robi>_<pod jakim_warunkiem>

+ bardziej generyczny warunek, for input 5 nie mówi za dużo

Reply:
Fakt to popoprawiam. 

Co do warunku to inne warunki są ciężkie typu for i = 0 : i = 15 itd. 

dlatego daje prosty int 5 jako pokazanie co zwróci dana klasa w tym przypadku Lexer.

Reply:
w senise to 5 to jest magiś magic number? czy to prostou dla zwykłęgo numeru zwraca EOF?